x-pack/filebeat/module/salesforce/logout-stream/test/salesforce_logout.ndjson.log-expected.json

```diff
@@ -0,0 +1,58 @@
+{

```

Comment:
Did you copy manually or generate `-expected.json` files?

Reply:
We generated the `-expected.json` file by running the pipeline after importing the processors and adding the sample document in the `Stack Management > Ingest Pipelines > Create Pipeline` section in Kibana. We have added the generated files out here.

Reply:
I see, but it isn't the right way to generate them. You should use Python integration tests to generate them. See this [comment](https://github.com/elastic/beats/pull/27623#issuecomment-910032458).

Reply:
I tried running the following command and faced an issue:

Command: `PYTEST_ADDOPTS="-k test_xpack_modules" GENERATE=1 TESTING_FILEBEAT_MODULES=salesforce mage pythonIntegTest`

Issue:
```
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
    <testsuite name="pytest" errors="0" failures="11" skipped="0" tests="11" time="116.958" timestamp="2021-12-17T09:55:04.390402" hostname="ec0cb65ff549">
        <testcase classname="x-pack.filebeat.tests.system.test_xpack_modules.XPackTest" name="test_fileset_file_00_salesforce" file="filebeat/tests/system/test_modules.py" line="77" time="10.512">
            <failure message="beat.beat.TimeoutError: Timeout waiting for 'cond' to be true. Waited 10 seconds.">
```
@kush-elastic was also facing this error before as well and we were trying to resolve it.
Are you aware of any specific reason why this could be happening?

Reply:
Did you check for any errors in the Docker logs?

Reply:
Let me drop a comment from @ChrsMark:

>Hey! This
Timeout waiting for 'cond' to be true. Waited 10 seconds.
message looks like the beats does not actually generate events that the test expects to find, no?
>In such cases Filebeat’s logs would help, and iirc they can find them under _build directory where the beats started by the tests actually log.

Reply:
Found this issue in the log. We analysed that it's the first parameter of the first fileset inside the module. Does this error is because it's running in our local machines because of some reason?
We were thinking of letting it run in the CI once and verifying if face the same issue there.
```
{
    "log.level": "error",
    "@timestamp": "2021-12-17T10:41:48.922-0200",
    "log.origin": {
        "file.name": "instance/beat.go",
        "file.line": 994
    },
    "message": "Exiting: error getting config for fileset salesforce/apex-rest: Error interpreting the template of the input: template: text:7:16: executing \"text\" at <.client_id>: map has no entry for key \"client_id\"",
    "service.name": "filebeat",
    "ecs.version": "1.6.0"
}
```